datashader/bokeh_ext.py

```diff
@@ -188,18 +208,35 @@ def update(self, ranges):
             Dictionary with of x/y-ranges, width and height.
         """
         if not self.comms_handle:
-            self.comms_handle = _CommsHandle(get_comms(self.ref),
-                                             self.doc, {})
+            comm = get_comms(self.ref)
+            comm_args = (comm, self.doc) if bokeh_version > '0.12.9' else (comm, self.doc, {})
+            self.comms_handle = CommsHandle(*comm_args)
 
         self.update_image(ranges)
         msg = self.get_update_event()
-        self.comms_handle.comms.send(msg)
+        comm = self.comms_handle.comms
+        if bokeh_version > '0.12.9':
+            comm.send(msg.header_json)
+            comm.send(msg.metadata_json)
+            comm.send(msg.content_json)
+            for header, payload in msg.buffers:
+                comm.send(json.dumps(header))
+                comm.send(buffers=[payload])
+        else:
+            comm.send(msg)
```

Comment:
Can you wrap this up a bit more simply, with the 0.12.9 version being encapsulated into a function to send a message, and then only switch between two versions of that function?  I think that would be easier for us to maintain.